mllib/src/main/scala/org/apache/spark/mllib/stat/MultivariateOnlineSummarizer.scala

```diff
@@ -231,9 +231,9 @@ class MultivariateOnlineSummarizer extends MultivariateStatisticalSummary with S
    */
   @Since("1.1.0")
   override def numNonzeros: Vector = {
-    require(totalWeightSum > 0, s"Nothing has been added to this summarizer.")
+    require(totalCnt > 0, s"Nothing has been added to this summarizer.")
 
-    Vectors.dense(weightSum)
+    Vectors.dense(nnz.map(_.asInstanceOf[Double]))
```

Comment:
The former is a cast, which is 'wrong' because a Long is not a Double, but maybe works out with implicits. I assume that's slower. Really you mean "take this long and convert it to a double" so the latter is better.